### PR TITLE
Hack register name for cortex-m and show flags.

### PR DESCRIPTION
**Pull request by douzepouze**
This hacky patch renames the cpsr ARM register to the actual xPSR name
when using arm-none-eabi-gdb.

It also decodes the most important ALU flags.

### CODE REVIEW

**Comment by cyrus-and** [comment]
I'd like to test it before so I need to set up a QEMU instance or something... in the meanwhile can you elaborate on why you need this? (I have basically no experience with the ARM architecture.)

File: .gdbinit [hunk 1]
```diff
@@ -1391,14 +1391,32 @@ class Registers(Dashboard.Module):
...
             self.table[name] = string_value
+
+            if name == "cpsr":
+                name = "xPSR"
+
+                indicator = []
+                int_value = int(string_value[2:], 16)
+                flags = { 31: 'N', 30: 'Z', 29: 'C', 28: 'V', 27: 'Q'}
+                for bit, flag in flags.items():
+                    bit_value = (1 << bit) & int_value != 0
```
Comment (JL2210) [inline .gdbinit]: This might be more obvious what it's doing if you write `bit_value = ((1 << bit) & int_value) != 0`

**Comment by cyrus-and** [comment]
> in the meanwhile can you elaborate on why you need this?

I still need clarification on this.

**Comment by cyrus-and** [comment]
Anyway this is a too niche feature that I don't feel like to add in the project. If your architecture requires special treatment you can always write a custom module for that.

**Pull request closed without merging**